src/rgw/rgw_rest.cc

```diff
@@ -1836,14 +1837,7 @@ int RGWHandler_REST::allocate_formatter(struct req_state *s,
 
 int RGWHandler_REST::validate_tenant_name(string const& t)
 {
-  struct tench {
-    static bool is_good(char ch) {
-      return isalnum(ch) || ch == '_';
-    }
-  };
-  std::string::const_iterator it =
-    std::find_if_not(t.begin(), t.end(), tench::is_good);
-  return (it == t.end())? 0: -ERR_INVALID_TENANT_NAME;
+  return rgw_is_valid_tenant(t);
```

Comment:
i think we're better off removing `RGWHandler_REST::validate_tenant_name` altogether, and changing any callers to use the new free function directly